apps/shell/app.js

```diff
@@ -180,6 +188,9 @@ function createLogger() {
       }
       logQueuedMessages();
     },
+    setpid: (newpid) => {
+      msgPrefix = `[User = ${username}; Host = ${host}; PID = ${newpid}]`;
+    },
```

Comment:
Is there any way to do this in a more functional style so we don't have `msgPrefix` as a variable?

Reply:
The challenge is that the PID isn't known when the logger is initialized. The PID will be known once the code gets to this line: 

https://github.com/OSC/ondemand/pull/4628/files#diff-e21ee4d7161757c9f1dbb11896bbcbc3c2749663d5f6153873a7e98ae7f20f10L214

```js
 term = pty.spawn(cmd, args, {
//...
```

Reply:
One solution is to log the `Web socket connection established from user` message with a normal `console.log()`, and initialize the `ws.logger` once the `pty-term` PID is known. I'll push an update.